### PR TITLE
Draft - Multiple routes support

### PR DESCRIPTION
**Pull request by lordrip**
Draft pull request to start the conversation about how to support multiple routes from the backend. The result of this conversation should shape the frontend effort.

### CODE REVIEW

File: api/src/main/java/io/kaoto/backend/api/resource/v2/IntegrationsResource.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+/**
+ * 🐱class IntegrationsResource
+ * 🐱relationship compositionOf DeploymentService, 0..1
+ *
+ * This endpoint will return the yaml needed to deploy
+ * the related integration and the
+ * endpoints to interact with deployments.
+ */
+@Path("/v2/integrations")
```
Comment (lordrip) [inline api/src/main/java/io/kaoto/backend/api/resource/v2/IntegrationsResource.java:32]: @Delawen I created a new resource to have a different path, to keep backward compatibility

File: services-interfaces/src/main/java/io/kaoto/backend/api/service/step/parser/StepParserService.java [hunk 2]
```diff
@@ -40,6 +40,16 @@
...
 
+    /*
+     * 🐱method getParsedFlows: List<ParseResult>
+     * 🐱param yaml: String
+     *
+     * Based on the YAML provided, offer a list of flows defined on it and
+     * the metadata associated to the orchestration.
+     */
+    @WithSpan
+    List<ParseResult<T>> getParsedFlows(String yaml);
```
Comment (lordrip) [inline services-interfaces/src/main/java/io/kaoto/backend/api/service/step/parser/StepParserService.java:51]: This is a new method to return multiple flows from a CRD, but I'm not sure if this is the right approach.

File: camel-route-support/src/main/java/io/kaoto/backend/api/service/step/parser/camelroute/CamelRouteStepParserService.java [hunk 3]
```diff
@@ -40,19 +40,29 @@ public String description() {
 
     @Override
     public ParseResult<Step> deepParse(final String input) {
+        // Right now we discard any flow that is not the first
+        return getParsedFlows(input).get(0);
+    }
+
+    @Override
+    public List<ParseResult<Step>> getParsedFlows(final String input) {
```
Comment (lordrip) [inline camel-route-support/src/main/java/io/kaoto/backend/api/service/step/parser/camelroute/CamelRouteStepParserService.java:48]: @Delawen, Initially, I was returning an array of Results, but it seems like the `Integration` class already supports multiple flows, I'm not sure about the `CamelRoute` class though.

That being said, I'm wondering if a better way would be to update the `ParseResult` class to support multiple flows instead. What do you think?

File: services-interfaces/src/main/java/io/kaoto/backend/api/service/step/parser/StepParserService.java (hunk 2 above)
Comment (Delawen) [reply services-interfaces/src/main/java/io/kaoto/backend/api/service/step/parser/StepParserService.java:51]: I guess it makes sense. We will deprecate the single flow return at some point.

**Comment by Delawen** [comment]
Continues in #638

**Pull request closed without merging**